expand/expand.go

```diff
@@ -215,6 +215,24 @@ func Pattern(cfg *Config, word *syntax.Word) (string, error) {
 func Format(cfg *Config, format string, args []string) (string, int, error) {
 	cfg = prepareConfig(cfg)
 	buf := cfg.strBuilder()
+
+	consumed, err := FormatIntoBuffer(buf, format, args)
+	if err != nil {
+		return "", 0, err
+	}
+
+	return buf.String(), consumed, err
+}
+
+// Format expands a format string with a number of arguments, following the
+// shell's format specifications. These include printf(1), among others.
+//
+// The resulting string is written to the provided buffer, and the number
+// of arguments used is returned.
+//
+// The config specifies shell expansion options; nil behaves the same as an
+// empty config.
+func FormatIntoBuffer(buf *bytes.Buffer, format string, args []string) (int, error) {
```

Comment:
do we need to expose this function right away? I imagine users will still be calling Format - if for any reason that API isn't performant enough, we can discuss a lower-level API as a separate topic. Exporting doesn't seem required for supporting %b.

Reply:
That's a good point. I've updated the PR to not export this function.